taraxa/state/dpos/precompiled/dpos_contract.go

```diff
@@ -468,43 +471,56 @@ func (self *Contract) Run(ctx vm.CallFrame, evm *vm.EVM) ([]byte, error) {
 			fmt.Println("Unable to parse getValidators input args: ", err)
 			return nil, err
 		}
-
-		res := self.getValidators(args)
-		return method.Outputs.Pack(res.Validators, res.End)
+		return method.Outputs.Pack(self.getValidators(args))
 
 	case "getDelegations":
 		var args sol.GetDelegationsArgs
 		if err = method.Inputs.Unpack(&args, input); err != nil {
 			fmt.Println("Unable to parse getDelegations input args: ", err)
 			return nil, err
 		}
-
-		res := self.getDelegations(args)
-		return method.Outputs.Pack(res.Delegations, res.End)
+		return method.Outputs.Pack(self.getDelegations(args))
 
 	case "getUndelegations":
 		var args sol.GetUndelegationsArgs
 		if err = method.Inputs.Unpack(&args, input); err != nil {
 			fmt.Println("Unable to parse getUndelegations input args: ", err)
 			return nil, err
 		}
-
-		res := self.getUndelegations(args)
-		return method.Outputs.Pack(res.Undelegations, res.End)
+		return method.Outputs.Pack(self.getUndelegations(args))
 	}
 
 	return nil, nil
 }
 
-func (self *Contract) DistributeRewards(rewardsStats *rewards_stats.RewardsStats, feesRewards *FeesRewards) {
+func (self *Contract) DistributeRewards(author *common.Address, rewardsStats *rewards_stats.RewardsStats, feesRewards *FeesRewards) {
 	// When calling DistributeRewards, internal structures must be always initialized
 	self.lazy_init()
 
 	// Calculates number of tokens to be generated as block reward
 	blockReward := bigutil.Mul(self.amount_delegated, self.yield_percentage)
 	blockReward = bigutil.Div(blockReward, bigutil.Mul(Big100, self.blocks_per_year))
 
+	voteReward := bigutil.Big0
+	authorReward := bigutil.Big0
+	trxReward := blockReward
+	// We need to handle case for block 0
+	if rewardsStats.TotalVotesCount > 0 {
+		// Calculate propotion between votes and transactions
+		voteReward = bigutil.Div(bigutil.Mul(blockReward, VotesToTrasnactionsRatio), Big100)
```

Comment:
if you use TransactionsToVotesRatio, which is 80 compared to 20, you should theoretically get smaller deviation after division

Reply:
+ it is votesReward

Reply:
I do not understand what is the difference between 20 and 80 maybe we should have a call about that. And voteReward is correct, as it will represents reward for one vote. This is different from trx because one validator can have only one vote

Reply:
- if you multiply some huge number by 80 and then divide it by 100, the rounding error will be smaller compared to multiplying it by 20 and dividing by 100
- see my other comments why you cannot calculate reward per 1 vote

Reply:
@JakubFornadel  where did you explain why I can not calculate reward per 1 vote? I can not see your comment